File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.c
#include <stdlib.h>

int cmp(const void *a, const void *b) {
    return (*(int *)a) - (*(int *)b);
}

int minimumCost(int* cost, int costSize)
{
    qsort(cost, costSize, sizeof(int), cmp);

    int sum = 0;
    int count = 0;

    for (int i = costSize - 1; i >= 0; i--) {
        count++;

        if (count % 3 != 0) {
            sum += cost[i];
        }
    }

    return sum;
}

File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.java
class Solution {
    public int minimumCost(int[] cost)
    {
        int sum=0,count=0;
        int n=cost.length;
     Arrays.sort(cost);
     for(int i=n-1;i>=0;i--)
     {
        if(count<2)
        {
            sum+=cost[i];
        }
         count++;
        if(count==3)
            count=0;
     }  
     return sum; 
    }
}